### PR TITLE
init auth tests

### PR DESCRIPTION
**Pull request by ShubhankarKG**
1. Route tests done.
2. Had to make some adjustments in the API, nothing breaking, yet.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "start": "node compiled/app.js",
     "dev": "nodemon --watch 'src/**/*.ts' --exec 'ts-node' src/app.ts",
     "pretest": "eslint . --ext .ts --fix",
-    "test": "cross-env NODE_ENV=test mocha -r ts-node/register 'tests/**/*.ts' --timeout 120000 --exit",
+    "test": "cross-env NODE_ENV=test mocha -r ts-node/register 'tests/**/*.ts' --timeout 24000 --exit",
+    "coverage": "nyc npm test",
```
Comment (ShubhankarKG) [inline server/package.json:13]: This outputs a coverage report, maybe we can upload it to codecov.io in a later PR.

File: server/src/app.ts [hunk 2]
```diff
@@ -13,16 +13,17 @@ dbConnect();
 app.use(express.json({ limit: "10mb" }));
 app.use(cors());
 
-if (process.env.NODE_ENV !== "test") {
-  app.use(morgan("combined"));
-}
+app.use(morgan("dev"));
```
Comment (BhavyaSheth22) [inline server/src/app.ts:16]: Can we disable morgan logging for the test env? The console logs with morgan are not very nice to see :)

File: server/src/controllers/auth.controller.ts [hunk 3]
```diff
@@ -84,8 +83,13 @@ const AuthController = {
           user: user,
           token: token
         });
+      } else {
+        return res.status(404).json({
+          error: "Requested user not found"
+        });
       }
     } catch (err) {
+      console.log(err);
```
Comment (BhavyaSheth22) [inline server/src/controllers/auth.controller.ts:92]: Please remove this log if not required.

File: server/tests/api/auth.test.ts [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+    describe("Bypassing VerifyOTP middleware", () => {
+        const email = "test-email@test.in";
+
+        let otp: number, hash: string;
+        beforeEach(() => {
+            [otp, hash] = generateOtpHash(email);
+        })
+        afterEach((done: Mocha.Done) => {
+            User.remove().then(() => done());
```
Comment (BhavyaSheth22) [inline server/tests/api/auth.test.ts:26]: .remove() has been deprecated, can we replace this by deleteOne() or deleteMany() perhaps?

File: server/tests/api/auth.test.ts [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+        describe('/POST login', () => {
+            it("logs in user correctly if user email is there", (done) => {
+                let user = new User({
+                    name: 'test user',
+                    email,
+                    mobileNumber: 9874563210,
+                    address: 'Chandni Chowk'
+                });
+                // const app = require("../../src/app");
```
Comment (BhavyaSheth22) [inline server/tests/api/auth.test.ts:82]: Please remove this inline comment.

**Review by BhavyaSheth22 — COMMENTED** [review]
Thanks a lot for this exhaustive testing! Just a few possible changes:

**Comment by BhavyaSheth22** [comment]
LGTM now! Thanks!

File: server/src/controllers/auth.controller.ts [hunk 6]
```diff
@@ -1,18 +1,17 @@
 import OTPUtil = require("./../utility/otp");
 import User from "./../models/User/User";
 import jwt from "jsonwebtoken";
-import dotenv from "dotenv";
 import express from "express";
+import IUser from "src/models/User/IUser";
 
-dotenv.config();
-const JWT_AUTH_TOKEN = process.env.JWT_AUTH_TOKEN || "";
+const JWT_AUTH_TOKEN = process.env.JWT_AUTH_TOKEN || "DSC_IS_GREAT";
```
Comment (pal-16) [inline server/src/controllers/auth.controller.ts:7]: :)

**Pull request merged**